### PR TITLE
Start incorporating Router

### PR DESCRIPTION
**Pull request by aspitz1**
*Don't merge this. Pull it down and make a new branch off of this branch to continue working on it, thank you!*
* Starts to incorporate Router
* Introduces lots of new bugs, the search feature needs to be completely refactored
* Home button works, but is active on the All-Movies view where it isn't really doing anything
* Moved the Movie fetch to the Movie component, converting it to a class component
* Movie cards need to become Links, which really messes up the styling big time!
I will add gif of what the site is looking like, including the error messages in the console when a movie is searched for by name
Rancid Tomatillos (6)
Screen Shot 2022-08-30 at 4 59 45 PM

### CODE REVIEW

File: src/All-Movies/All-Movies.js [hunk 1]
```diff
@@ -2,15 +2,14 @@ import React from 'react'
...
-const AllMovies = ({movies, switchView}) => {
+const AllMovies = ({movies}) => {
     const moviePosters = movies.map(movie => {
         return (
             <MovieCard
                 key={movie.id}
                 id={movie.id} 
                 title={movie.title}
                 poster={movie.poster_path}
-                switchView={switchView}
```
Comment (aspitz1) [inline src/All-Movies/All-Movies.js]: No longer using this method to switch views

File: src/App/App.js [hunk 2]
```diff
@@ -57,71 +33,17 @@ class App extends Component {
...
-          runtime: data.movie.runtime,
-          tagline: data.movie.tagline,
-          averageRating: data.movie['average_rating']
-        }})
-      })
-      .catch(err =>{
-        console.log(err)
-        this.setState({ ...this.state, error: 'Looks like something went wrong.' })
-      })
-  }
```
Comment (aspitz1) [inline src/App/App.js:33]: Moved everything that handles fetching a movie to the Movie component, which is now a class component

File: src/App/App.js [hunk 3]
```diff
@@ -57,71 +33,17 @@ class App extends Component {
...
         <Navbar 
           view={this.state.isMovieView} 
-          goHome={ this.goHome }
           findMovieByTitle={this.findMovieByTitle}
         />
         {this.state.error && <h3 className='error'>{this.state.error}</h3>}
-        {this.state.isMovieView || <AllMovies movies={this.state.movies} switchView={this.viewMovie}/>}
-        {this.state.isMovieView && <Movie movieDetails={this.state.movie}/>}
+          <Route exact path='/' render={() => <AllMovies movies={this.state.movies}/>}/>
+          <Route exact path='/:id' render={({ match }) => <Movie id={match.params.id} />}/>
```
Comment (aspitz1) [inline src/App/App.js:55]: used the `match` to collect the movie `id`, when the Movie component loads it uses the prop (id) passed in to fetch the movie

File: src/Movie-Card/Movie-Card.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import React from 'react'
+import { Link } from 'react-router-dom';
 
-const MovieCard = ({ id, title, poster, switchView }) => {
-    return <img key={id} src={poster} id={id} name={title} alt={title} className='movie-card' height='300' width='200' onClick={() => switchView(id)}/>
+const MovieCard = ({ id, title, poster }) => {
+    return <Link to={'/' + id}><img key={id} src={poster} id={id} name={title} alt={title} className='movie-card' height='300' width='200'/></Link>
```
Comment (aspitz1) [inline src/Movie-Card/Movie-Card.js]: The movies need to be Links as far as I can tell in order to go to the Movie view.

File: src/Navbar/Navbar.js [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 
-const Navbar = ({ view, goHome, findMovieByTitle }) => {
+const Navbar = ({ findMovieByTitle }) => {
     return (
         <nav className='navbar'>
             <h1>Rancid Tomatillos</h1>
-            {view && <button className='navbar-home' onClick={goHome}>Home</button>}
-            {view || <SearchByNameForm findMovieByTitle={findMovieByTitle} />}
+            <NavLink to='/' className='navbar-home'>Home</NavLink>
+            <SearchByNameForm findMovieByTitle={findMovieByTitle} />
```
Comment (aspitz1) [inline src/Navbar/Navbar.js]: I am not sure how to make the Home button only work on the Movie page and the Search button work on the All Movie page. We may be able to make them both make sense on work on both pages. Maybe just disable the home button on the home page?

File: src/Navbar/Navbar.scss [hunk 6]
```diff
@@ -33,6 +33,10 @@
     }
 }
 
+.navbar-home {
+    text-decoration: none;
```
Comment (aspitz1) [inline src/Navbar/Navbar.scss:38]: This gets rid of the default style the NavLink gives the button

File: src/Movie-Card/Movie-Card.js (hunk 4 above)
Comment (mayakappen) [reply src/Movie-Card/Movie-Card.js]: Do you think navlink might be easier or would it make a difference?

File: src/Navbar/Navbar.js (hunk 5 above)
Comment (mayakappen) [reply src/Navbar/Navbar.js]: Yeah do you think maybe an inline style on it on the home page to make it invisible?

**Comment by aspitz1** [comment]
The branch I pushed up and opened a PR on will hopefully resolve the conflicts on this branch as I merged this branch into it. The conflicts with this branch and the main were rather hard to solve and I ended up with a bunch of bugs afterwords. So fingers crossed merging the PR with the card styling fixes them!

File: src/Movie-Card/Movie-Card.js (hunk 4 above)
Comment (aspitz1) [reply src/Movie-Card/Movie-Card.js]: This ended up not being an issue with the code you wrote and I continued working on :)

File: src/Navbar/Navbar.js (hunk 5 above)
Comment (aspitz1) [reply src/Navbar/Navbar.js]: This is the issue you solved that you msged me about last night, right?

**Comment by aspitz1** [comment]
This is ready to merge in when you're ready @mayakappen :) I pulled it all down and tested it, this shouldn't break in the main branch

File: src/All-Movies/All-Movies.css [hunk 7]
```diff
@@ -1,26 +1,11 @@
 .movies-container {
   display: grid;
-  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
-  grid-gap: 10px;
+  grid-template-columns: repeat(auto-fill, minmax(189px, 1fr));
```
Comment (mayakappen) [inline src/All-Movies/All-Movies.css:3]: looks better with auto-fill!

File: src/Movie-Card/Movie-Card.js [hunk 8]
```diff
@@ -1,24 +1,21 @@
...
-    const releaseDate = new Date(release_date).toLocaleDateString('en-US')
-    const avgRating = average_rating.toFixed(1)
-    const backgroundImage = poster_path;
-    return <article key={id} id={id} name={title} alt={title} className='movie-card' onClick={() => switchView(id)} 
-    style={{height: '300px', width: '200px', backgroundImage: `url(${backgroundImage})`,backgroundRepeat: 'no-repeat',backgroundSize: 'contain',}}>
-                <div className='hidden'>
+const MovieCard = ({ id, title, poster, averageRating, releaseDate }) => {
+    const formattedReleaseDate = new Date(releaseDate).toLocaleDateString('en-US')
+    const avgRating = averageRating.toFixed(1)
+    return <Link to={'/' + id} id={id} name={title} alt={title} className='movie-card'
```
Comment (mayakappen) [inline src/Movie-Card/Movie-Card.js:8]: love how this works as a link :)

File: src/Movie-Card/Movie-Card.scss [hunk 9]
```diff
@@ -5,26 +5,26 @@
...
+    height: 300px;
+    width:  189px;
+    transition: all .2s ease-in-out;
     &:hover {
         cursor: pointer;
-        height: calc(100% + 20px);
-        width: calc(100% + 14px);
-        margin: 10px;
+        z-index: 1;
+        transform: scale(1.25);
```
Comment (mayakappen) [inline src/Movie-Card/Movie-Card.scss:21]: obsessed with transform rn this is so cool

**Review by mayakappen — COMMENTED** [review]
So much good stuff in here!! I love how we got the router stuff working :)

**Pull request merged**